torch/sparse/_triton_ops.py

```diff
@@ -77,6 +77,11 @@ def is_compatible_blocksize(b):
 
 
 def make_triton_contiguous(t):
+    # TODO: Why do we need "triton contiguity" that is not defined by
+    # triton itself? It looks like it is required until
+    # openai/triton#1291 fixed a bug in processing triton kernel
+    # arguments. Unless triton comntiguity is required for
+    # performance, remove this function.
```

Comment:
It is required for performance. See https://github.com/openai/triton/pull/852#issuecomment-1305688524

Reply:
Thanks, @nikitaved , for this information!

So, there are two separate problems:
1. Dealing with non-contiguous inputs having no strides equal to 1.
2. Applying `make_triton_contiguous` to `bsr_dense_mm`/`bsr_dense_addmm` output tensor that will trigger unnecessary copy and cause the bug described in the PR description.

Re the problem 1: with non-contiguous inputs, we can either (i) make inputs contiguous or (ii) leave as they are. Both options has costs. The option (i) involves several kernel calls (copy operations from applying `make_triton_contiguous` to non-contiguous inputs plus a `bsr_dense_mm`/`bsr_dense_addmm` kernel call). The option (ii) has a single kernel call but there may be a performance decrease per https://github.com/openai/triton/pull/852#issuecomment-1305688524.
It is not clear atm which of the two options have bigger impact to performance decrease. My gut feeling is (ii) due to several kernel calls. However, both options need to be tested before the final decision.

Re the problem 2: to solve this, we'll need to create the output tensor with predefined strides such that `make_triton_contiguous` will never trigger a copy. Or, undo https://github.com/pytorch/pytorch/pull/108512 .